### PR TITLE
Refactor EmptyGridWorld

### PR DESCRIPTION
**Pull request by Sid-Bhatia-0**
Refactoring `EmptyGridWorld` to make it more configurable:

1. Add keyword argument to configure goal position instead of hardcoding it.
2. Add keyword arguments to `reset!` method to allow configuring goal position, agent's starting position, and its direction from outside (through wrapper environments, for example).
3. Reuse the `reset!` method inside the environment constructor. As a general convention, we should probably have the constructor return an environment that has been `reset!`.

### CODE REVIEW

File: src/envs/emptygridworld.jl [hunk 1]
```diff
@@ -8,17 +8,21 @@ mutable struct EmptyGridWorld <: AbstractGridWorld
...
     w[WALL, 1:n, [1,n]] .= true
-    w[GOAL, n-1, n-1] = true
-    w[EMPTY, n-1, n-1] = false
     goal_reward = 1.0
     reward = 0.0
-    EmptyGridWorld(w, agent_start_pos, Agent(dir=agent_start_dir), goal_reward, reward)
+
+    env = EmptyGridWorld(w, agent_start_pos, Agent(dir=agent_start_dir), goal_reward, reward)
+
+    reset!(env, agent_start_pos = agent_start_pos, agent_start_dir = agent_start_dir, goal_pos = goal_pos)
```
Comment (findmyway) [inline src/envs/emptygridworld.jl:22]: I think there's no need to call `reset!` here?

**Review by findmyway — CHANGES_REQUESTED** [review]
Could you fix the error in Travis CI first?

File: src/envs/emptygridworld.jl (hunk 1 above)
Comment (Sid-Bhatia-0) [reply src/envs/emptygridworld.jl:22]: >Could you fix the error in Travis CI first?

Sorry, I do not know how to comprehend Travis CI 😅 

>I think there's no need to call reset! here?

We will have to set the goal and empty accordingly, right?

Comment (findmyway) [reply src/envs/emptygridworld.jl:22]: See https://travis-ci.com/github/JuliaReinforcementLearning/GridWorlds.jl/jobs/434229835#L296

> We will have to set the goal and empty accordingly, right?

I see. Sorry I missed it.

Comment (Sid-Bhatia-0) [reply src/envs/emptygridworld.jl:22]: Thanks for the link. Fixed the bug.

**Pull request merged**